modules/compute/src/main/java/org/apache/ignite/internal/compute/queue/QueueEntry.java

```diff
@@ -71,6 +96,31 @@ CompletableFuture<R> toFuture() {
         return future;
     }
 
+    /**
+     * Sets interrupt status of the worker thread.
+     */
+    public void interrupt() {
+        // Interrupt under the lock to prevent interrupting thread used by the pool for another task
+        lock.lock();
```

Comment:
I think `AtomicReference` can be a simpler solution for the problem you are trying to solve with the lock.